src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -363,30 +364,37 @@ class QueryCompiler(cq : ConjunctiveQuery, ss: CompilationState) {
         resolved
       }
       case BinaryOpExpr(lhs, op, rhs) => {
-        val resovledLhs = compileExpr(lhs)
-        val resovledRhs = compileExpr(rhs)
-        s"(${resovledLhs} ${op} ${resovledRhs})"
+        val resovledLhs = compileExpr(lhs, level + 1)
+        val resovledRhs = compileExpr(rhs, level + 1)
+        val sql = s"${resovledLhs} ${op} ${resovledRhs}"
+        if (level == 0) sql else s"(${sql})"
```

Comment:
Seems the only use for `level` here is to remove a pair of parenthesis at the top level (when `level` = 0). Maybe I'm missing something here, why is this necessary?